azurerm/internal/services/containers/kubernetes_cluster_validate.go

```diff
@@ -39,6 +39,23 @@ func validateKubernetesCluster(d *schema.ResourceData, cluster *containerservice
 		}
 	}
 
+	// ensure conditionally-required identity values are valid
+	if v, exists := d.GetOk("identity"); exists {
+		rawIdentity := v.([]interface{})
+
+		if len(rawIdentity) != 0 {
+			identity := rawIdentity[0].(map[string]interface{})
+
+			if identityType := identity["type"].(string); identityType == string(containerservice.ResourceIdentityTypeUserAssigned) {
+				userAssignedIdentityId := identity["user_assigned_identity_id"].(string)
+
+				if userAssignedIdentityId == "" {
+					return fmt.Errorf("when `identity.type` is UserAssigned then `user_assigned_identity_id` must be set")
+				}
+			}
+		}
+	}
```

Comment:
since this value is likely to be interpolated, unfortunately this won't work - there's a bug in CustomizeDiff where interpolated values are returned as empty strings, so this'll get triggered if this is interpolated (which it's highly likely to be) - as such could we remove this?

Reply:
https://github.com/terraform-providers/terraform-provider-azurerm/pull/10188

Opened another PR as I assume it should be slightly easier for you to pull in, 

otherwise just cherry-pick https://github.com/terraform-providers/terraform-provider-azurerm/pull/10188/commits/e2c92cd38803c670331d285817a4762743bd0033 into this branch

Reply:
cherry-picked e2c92cd38803c670331d285817a4762743bd0033 over to this branch